### PR TITLE
Copy files into / out of docker container

### PR DESCRIPTION
**Pull request by Paula-Kli**
Checking whether using the inbuild docker commands to copy files fixes the permission error when trying to copy the diagram cache to the host system

### CODE REVIEW

**Comment by Wuestengecko** [comment]
Not sure what's going on with mypy. Seems unrelated to this PR, I'll take a look into it.

File: capellambse/_native.py [hunk 1]
```diff
@@ -144,28 +148,86 @@ def __call__(
...
-            cmd,
-            stdin=subprocess.DEVNULL,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.STDOUT,
+    ):
+        entrypoint = self.get_entrypoint()
+        container_name = "".join(
+            random.choice(string.ascii_lowercase) for i in range(5)
+        )
+        with subprocess.Popen(
```
Comment (Wuestengecko) [inline capellambse/_native.py:156]: Style improvement: It's a bit inconsistent to have some commands be direct `subprocess.Popen / run` calls here and others be methods which wrap them. With the length of a subprocess call, IMO it makes most sense to extract all of them out, which should make this `__call__` method much more pleasant to read.

File: capellambse/_native.py [hunk 2]
```diff
@@ -144,28 +148,86 @@ def __call__(
...
+                    project_folder,
+                    *args,
+                ],
+                check=True,
+            )
+
+            self.copy_files_from_container(
+                project_folder, project, container_name
+            )
+            running_process.terminate()
```
[the remark this answers is not in the document]
Comment (Paula-Kli) [reply capellambse/_native.py:196]: Giving also windows users a chance to let the process clean up after itself is a good point. Nevertheless, I used `running_process.stdin.close()` which works and does not give any errors but the static code analysis from mypy (?) failes with the error `Item "None" of "Optional[IO[bytes]]" has no attribute "close" [union-attr]` - according to the [documentation](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.stdin) stdin only is not none if it was set to `PIPE` but I would not know any other reason for the error than that the static code analysis is not able to figure that out.
Would you be fine using `terminate` instead or have another idea?

File: capellambse/_native.py [hunk 3]
```diff
@@ -144,28 +148,86 @@ def __call__(
...
+                    _ARG_WORKSPACE,
+                    "/workspace",
+                    _ARG_PROJECT,
+                    project_folder,
+                    *args,
+                ],
+                check=True,
+            )
+
+            self.copy_files_from_container(
```
[the remark this answers is not in the document]
Comment (Paula-Kli) [reply capellambse/_native.py:193]: I don't think I completely get your question but I think it just copies whatever files are present as written [here](https://docs.docker.com/engine/reference/commandline/cp/#:~:text=The%20cp%20command%20behaves%20like%20the%20Unix%20cp%20%2Da%20command%20in%20that%20directories%20are%20copied%20recursively%20with%20permissions%20preserved%20if%20possible.) it  behaves like the unix command `cp -a` copying recursively

[the remark this answers is not in the document]
Comment (Wuestengecko) [reply capellambse/_native.py:193]: That's what I was asking, thanks for clarifying :)

This is fine for now, nothing else do be done then.

File: capellambse/_native.py [hunk 4]
```diff
@@ -144,28 +147,99 @@ def __call__(
...
+            ["docker", "inspect", self.image],
+            check=True,
+            capture_output=True,
+        ).stdout
+        decoded_output = output.decode("utf-8")
+        parsed_output = json.loads(decoded_output)[0]
+        return parsed_output["Config"]["Entrypoint"]
+
+    def start_container(self, container_name: str) -> subprocess.Popen:
+        # pylint: disable-next=consider-using-with
```
[the remark this answers is not in the document]
Comment (Paula-Kli) [reply capellambse/_native.py]: ^^ I implemented it that way now

File: capellambse/_native.py [hunk 5]
```diff
@@ -144,28 +147,99 @@ def __call__(
...
-        _LOGGER.debug("Running Capella in Docker with command %r", cmd)
-        return subprocess.run(
-            cmd,
-            stdin=subprocess.DEVNULL,
+    ):
+        entrypoint = self.get_entrypoint()
+        container_name = "capellambse-" + "".join(
+            random.choice(string.ascii_lowercase) for i in range(5)
+        )
+        running_process = self.start_container(container_name)
```
[the remark this answers is not in the document]
Comment (Paula-Kli) [reply capellambse/_native.py]: this is already done with https://github.com/DSD-DBS/py-capellambse/pull/309#discussion_r1263461178 correct?

**Pull request merged**